src/platform/site-wide/user-nav/containers/AutoSSO.jsx

```diff
@@ -8,42 +12,75 @@ import {
 } from 'platform/user/authentication/selectors';
 import { autoLogin, autoLogout } from 'platform/user/authentication/utilities';
 import { hasSession, hasSessionSSO } from 'platform/user/profile/utilities';
-import { ssoKeepAliveSession } from 'platform/utilities/sso';
+import {
+  ssoKeepAliveSession,
+  getForceAuth,
+  setForceAuth,
+  deleteForceAuth,
+} from 'platform/utilities/sso';
+
+function parseqs(value) {
+  /*
+   * naive query string parsing function, takes a query string and returns
+   * an object mapping the keys to values.
+   */
+  const data = value.startsWith('?') ? value.substring(1) : value;
+  const entries = data ? data.split('&').map(q => q.split('=')) : [];
+  return fromPairs(entries);
+}
 
 export async function checkStatus(toggleKeepAlive) {
   await ssoKeepAliveSession();
   if (hasSession() && !hasSessionSSO()) {
     autoLogout();
-  } else if (!hasSession() && hasSessionSSO()) {
+  } else if (!hasSession() && hasSessionSSO() && !getForceAuth()) {
     autoLogin();
   }
 
   toggleKeepAlive();
 }
 
-function AutoSSO(props) {
-  const { useSSOe, useInboundSSOe, hasCalledKeepAlive } = props;
+class AutoSSO extends React.Component {
+  render() {
+    const {
+      useSSOe,
+      useInboundSSOe,
+      hasCalledKeepAlive,
+      userLoggedIn,
+    } = this.props;
 
-  if (useSSOe && useInboundSSOe && !hasCalledKeepAlive) {
-    checkStatus(props.checkKeepAlive);
-  }
+    if (parseqs(window.location.search).auth === 'force-needed') {
+      setForceAuth();
+    }
 
-  return null;
+    if (userLoggedIn) {
+      deleteForceAuth();
+    }
+
+    if (useSSOe && useInboundSSOe && !hasCalledKeepAlive) {
+      checkStatus(this.props.checkKeepAlive);
+    }
+
+    return null;
+  }
 }
 
 const mapStateToProps = state => ({
   useSSOe: ssoe(state),
   useInboundSSOe: ssoeInbound(state),
   hasCalledKeepAlive: hasCheckedKeepAlive(state),
+  userLoggedIn: isLoggedIn(state),
 });
 
 const mapDispatchToProps = {
   checkKeepAlive,
 };
 
-export default connect(
-  mapStateToProps,
-  mapDispatchToProps,
-)(AutoSSO);
+export default withRouter(
```

Comment:
I don't think you need `withRouter` anymore, right?

Reply:
nope, thanks for catching